Feature/form

Luminilion opened the pull request:
Contact event organizers through registered event mail.
Contact form is reachable from EventMainFragment through button under description.
Form will let user choose automatically his email app or choose default one.

Commit:
Included Button to go to form

Commit:
Added email field in event

Commit:
Added button in xml code

Commit:
Created form fragment and xml code + added strings in xml file

zyuiop commented on app/src/main/java/ch/epfl/sweng/eventmanager/repository/data/Event.java:
```diff
@@ -34,6 +34,11 @@
+    /**
+     * An email to contact the event organizers
+     */
+    private String email;
```
Please add the email to the constructor, and update the tests accordingly.

zyuiop replied:
Actually: remove this field, and add it to the EventOrganizer

zyuiop commented on app/src/main/java/ch/epfl/sweng/eventmanager/ui/eventShowcase/fragments/EventFormFragment.java:
```diff
@@ -0,0 +1,90 @@
+    private final int REQUEST_CODE = 1;
+
+    @BindView(R.id.contact_form_send_button)
+    Button send_button;
```
Please respect java naming conventions!

```suggestion
    Button sendButton;
```

zyuiop commented on app/src/main/java/ch/epfl/sweng/eventmanager/ui/eventShowcase/fragments/EventFormFragment.java:
```diff
@@ -0,0 +1,90 @@
+        model.getEvent().observe(this, ev -> {
+            email = ev.getEmail();
+
+            send_button.setOnClickListener(v -> {
```
```suggestion
            sendButton.setOnClickListener(v -> {
```

zyuiop commented on app/src/main/java/ch/epfl/sweng/eventmanager/ui/eventShowcase/fragments/EventFormFragment.java:
```diff
@@ -0,0 +1,90 @@
+    }
+
+    @Override
+    public void onResume() {
```
Method is too long, split it (and it will be easier to test :p)

zyuiop commented on app/src/main/java/ch/epfl/sweng/eventmanager/ui/eventShowcase/fragments/EventMainFragment.java:
```diff
@@ -34,6 +36,15 @@ public void onResume() {
 
             TextView eventDescription = (TextView) view.findViewById(R.id.event_description);
             eventDescription.setText(ev.getDescription());
+            Button contactButton = view.findViewById(R.id.contact_form_go_button);
```
We may want to put this in the lateral menu

Luminilion replied:
I think for now I would put it here, this can be easely changed in the future
Anyway tests do not pass

Luminilion replied:
@zyuiop if you want the button to be displayed elsewhere please create an issue so that I can merge this one and keep thing simple :) 
It is currently only a way of accessing the form and certainly is not its final place !

zyuiop reviewed (changes requested):
Some quick changes to make, and don't forget to write tests.
Also, don't forget to update the data on firebase to have some emails we can test with :)

Commit:
Created test file for formFragment

Test file is not working because onView:47 does not find a match
Found no solution to that

zyuiop committed:
Update app/src/main/java/ch/epfl/sweng/eventmanager/ui/eventShowcase/fragments/EventFormFragment.java

Co-Authored-By: Luminilion <nico.darge@gmail.com>

zyuiop committed:
Update app/src/main/java/ch/epfl/sweng/eventmanager/ui/eventShowcase/fragments/EventFormFragment.java

Co-Authored-By: Luminilion <nico.darge@gmail.com>

Luminilion committed:
Merge branch 'master' into feature/form

Commit:
Used method changeFragment from EventShowcaseActivity

Method is used in EventMainFragment to go to form fragment
Tests do not pass
Removed unused imports in EventFormFragment
Removed findById casts as they are done automatically

Commit:
Refactored a method + changed error for empty fields

Created method to check if EditText were correctly filled
Error is now displayed as an EditText.setError message

Commit:
Changed Intent RequestCode and standardised error message

Changed Code to Activity.RESULT_OK (=-1, android official request code)
(before was a handmade request code)
Set Focus on empty field if any
Standardized all empty field error to R.string.contact_form_error

Commit:
Made FormTest work by getting to fragment in @Before + added util class

ToastMatcher is not used here but can be very useful in case of Toast
testing, how to use the class is in the class doc.
To be deleted if judged irrelevant.

Commit:
Refactored EventFormFragment to have less code lines in onResume()

Luminilion committed:
Merge branch 'master' into feature/form

Luminilion committed:
Merge branch 'master' into feature/form

Commit:
Moved email field to eventOrganizer class + included in constructor

Commit:
Modified where to get email when event is available

Commit:
Merge remote-tracking branch 'origin/feature/form' into feature/form

# Conflicts:
#	app/src/main/java/ch/epfl/sweng/eventmanager/repository/data/Event.java

Commit:
Changed onIdle() import from androidx to android.support

Commit:
Changed EventOrganizer instantiation

Luminilion committed:
Merge branch 'master' into feature/form

Commit:
Modified constructor for Event in EventTest to include Twitter Name

Commit:
Changed string format in intent.EXTRA_MAIL to have it correctly read
